File: src/__tests__/enums_test.py
import unittest
from utils import enums
import os

HERE = os.path.dirname(os.path.abspath(__file__))

filepath = os.path.join(HERE, "test_data", "enum.smali")
test_file = open(filepath, 'r')
lines = test_file.readlines()
content = ''.join(lines)
test_file.close()


class EnumsTest(unittest.TestCase):
    def test_is_enum_file(self):
        self.assertEqual(enums.get_enum_name(
            '.class public final enum Lcom/reddit/mod/temporaryevents/models/TemporaryEventFields$TempEventBoolean;\n'),  # noqa
            'Lcom/reddit/mod/temporaryevents/models/TemporaryEventFields$TempEventBoolean;'  # noqa
            )

    def test_extract_enum_values(self):
        self.assertEqual(enums.extract_enum_values(
            lines), ['FALSE', 'NONE', 'TRUE'])


if __name__ == '__main__':
    unittest.main()

File: internal/utils/enums_test.go
package utils

import (
	"bufio"
	"os"
	"slices"
	"testing"
)

func TestGetEnumName(t *testing.T) {
	f, err := os.Open("testdata/enum.smali")
	if err != nil {
		t.Error(err)
	}
	scanner := bufio.NewScanner(bufio.NewReader(f))
	lines := []string{}
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	res := GetEnumName(lines[0])
	if !res.IsOk {
		t.Error()
	}

	if *res.Value != "Lcom/reddit/mod/temporaryevents/models/TemporaryEventFields$TempEventBoolean;" {
		t.Error()
	}
}

func TestExtractEnumValues(t *testing.T) {
	f, err := os.Open("testdata/enum.smali")
	if err != nil {
		t.Error(err)
	}
	scanner := bufio.NewScanner(bufio.NewReader(f))
	lines := []string{}
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	enums := ExtractEnumValues(lines)

	if !slices.Equal(enums, []string{"FALSE", "NONE", "TRUE"}) {
		t.Error()
	}
}
